registry/proxy/proxyregistry.go

```diff
@@ -274,8 +274,9 @@ func (r *remoteAuthChallenger) tryEstablishChallenges(ctx context.Context) error
 	if err := ping(r.cm, remoteURL.String(), challengeHeader); err != nil {
 		return err
 	}
+	dcontext.GetLogger(ctx).Infof("Challenge established with upstream: %+v", remoteURL)
+	dcontext.GetLogger(ctx).Infof("Challenge manager: %+v", r.cm)
```

Comment:
Not sure if logging the challenge manager is producing anything useful; at least, looking at the code;
https://github.com/distribution/distribution/blob/4d613f771f4565f0ab2eaaf14ce34037ab033e2c/registry/proxy/proxyregistry.go#L245


And `challenge.Manager` looking like this;
https://github.com/distribution/distribution/blob/a4918b67bbe4f2bdf870bb6fad345529555f107e/internal/client/auth/challenge/authchallenge.go#L68-L80

Given that `remoteURL` is just an URL, perhaps instead;



```suggestion
	dcontext.GetLogger(ctx).Infof("Challenge established with upstream: %s", remoteURL.String())
```

Or better; if we expect the URL to potentially include `username:password`;




```suggestion
	dcontext.GetLogger(ctx).Infof("Challenge established with upstream: %s", remoteURL.Redacted())
```